### PR TITLE
Update contributing

### PR DESCRIPTION
**Pull request by protoroto**
## Description

- Update contributing section
- Minor style fixes

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -46,3 +46,121 @@ If you are proposing a feature:
...
+as soon as possible.
+
+### Get Started!
+
+Ready to contribute? Refer to the project `Get Started!` section of `CONTRIBUTING.rst`
+to set up your environment for local development. 
+
+We use, for most projects, this set of packages:
+
+- `bumpversion`_
```
Comment (yakky) [inline contributing.md]: links must be updated to use markdown syntax, original document is in REST format

File: contributing.md [hunk 2]
```diff
@@ -46,3 +46,121 @@ If you are proposing a feature:
...
+You may want to install them in the global environment::
+
+    pip3 install --user bump2version towncrier invoke tox
+
+#### Development tips
+
+Most of the projects allows you to use `pre-commit <https://pre-commit.com/>`_ to ensure an easy compliance
+to the project code styles.
+
+If you want to use it, install it globally (for example with `pip3 install --user precommit`,
```
Comment (yakky) [inline contributing.md]:
```suggestion
If you want to use it, install it globally (for example with `pip3 install --user pre-commit`,
```

**Review by yakky — CHANGES_REQUESTED** [review]

File: contributing.md (hunk 1 above)
Comment (protoroto) [reply contributing.md]: @yakky Should be fixed now

File: contributing.md (hunk 2 above)
Comment (protoroto) [reply contributing.md]: @yakky Right. Fixed

File: contributing.md [hunk 3]
```diff
@@ -46,3 +46,116 @@ If you are proposing a feature:
...
+   one line. It will be added automatically to the history file upon release.
+   File must be named as `<issue-number>.<type>` with type being:
+
+   * `.feature`: For new features.
+   * `.bugfix`: For bug fixes.
+   * `.doc`: For documentation improvement.
+   * `.removal`: For deprecation or removal of public API.
+   * `.misc`: For general issues.
+
+   Check `towncrier`_ documentation for more details.
```
Comment (yakky) [inline contributing.md]: still needs fix

**Review by yakky — CHANGES_REQUESTED** [review]

File: contributing.md (hunk 3 above)
Comment (protoroto) [reply contributing.md]: @yakky Sorry, missed it. Fixed :)

**Review by yakky — APPROVED** [review]

**Pull request merged**